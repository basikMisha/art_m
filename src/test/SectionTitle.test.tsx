import { render, screen } from '@testing-library/react';
import SectionTitle from '@/components/Title';
import { describe, it, expect } from 'vitest';

describe('SectionTitle', () => {
  it('отображает переданный заголовок и подзаголовок', () => {
    render(<SectionTitle title="Главный заголовок" subtitle="Подзаголовок" />);

    expect(screen.getByText('Главный заголовок')).toBeInTheDocument();
    expect(screen.getByText('Подзаголовок')).toBeInTheDocument();
  });

  it('корректно рендерится без ошибок', () => {
    const { container } = render(<SectionTitle title="Test Title" subtitle="Test Subtitle" />);
    expect(container).toMatchSnapshot();
  });

  it('поддерживает динамические данные', () => {
    const title = 'Динамический заголовок';
    const subtitle = 'Динамический подзаголовок';

    render(<SectionTitle title={title} subtitle={subtitle} />);

    expect(screen.getByText(title)).toBeInTheDocument();
    expect(screen.getByText(subtitle)).toBeInTheDocument();
  });
});
