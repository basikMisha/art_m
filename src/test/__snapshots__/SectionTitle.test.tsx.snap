// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SectionTitle > корректно рендерится без ошибок 1`] = `
<div>
  <div
    class="sc-KXCwU dmAaJW"
  >
    <p
      class="sc-gyycJP fBxZrG"
    >
      Test Subtitle
    </p>
    <h2
      class="sc-eaUbBy bAedUc"
    >
      Test Title
    </h2>
  </div>
</div>
`;
